prefect_census/client.py

```diff
@@ -0,0 +1,93 @@
+"""Module containing client for interacting with the Census API"""
+from httpx import AsyncClient, Response
+
+
+class CensusClient:
+    """
+    Client for interacting with the Census API.
+
+    Args:
+        api_key: API key to authenticate with the Census API.
+    """
+
+    def __init__(self, api_key: str):
+        self._closed = False
+        self._started = False
+
+        self.client = AsyncClient(
+            base_url="https://app.getcensus.com",
+            headers={"Authorization": f"Bearer {api_key}"},
+        )
+
+    async def call_endpoint(
+        self,
+        http_method: str,
+        path: str,
+    ) -> Response:
+        """
+        Call an endpoint in the Census API.
+
+        Args:
+            http_method: HTTP method to call on the endpoint.
+            path: The partial path for request (e.g. //api/v1/syncs/42). Will be
+            appended onto the base URL as determined by the client configuration.
+
+        Returns:
+            The response from the Census API.
+        """
+
+        response = await self.client.request(
+            method=http_method,
+            url=path,
+        )
+        response.raise_for_status()
+        return response
+
+    async def get_run_info(self, run_id: int):
+        """
+        Sends a request to the [get sync id info endpoint]
+        (https://docs.getcensus.com/basics/api/syncs#get-syncs-id)
+
+        Args:
+            run_id: The ID of the sync run to get details for.
+
+        Returns:
+            The response from the Census API.
+        """  # noqa
+        return await self.call_endpoint(
+            http_method="GET", path=f"/api/v1/sync_runs/{run_id}"
+        )
+
+    async def trigger_sync_run(self, sync_id: int):
```

Comment:
We should include a `force_full_sync` parameter to ensure that we're supporting all the functionality that the Census API has to offer.
```suggestion
    async def trigger_sync_run(self, sync_id: int, force_full_sync=False):
```